src/Shabalov/Task-5/Plane.cpp

```diff
@@ -0,0 +1,15 @@
+#include "Plane.h"
+#include <utility>
+
+Plane::Plane(const QVector3D &p, const QVector3D &n, Material m) : position(p), normal(n), material(std::move(m)){
+}
+
+bool Plane::ray_intersect(const Ray &ray, float &t) const {
+    auto denom = QVector3D::dotProduct(-normal, ray.direction);
```

Comment:
```suggestion
    const auto denom = QVector3D::dotProduct(-normal, ray.direction);
```

все что можно сделать константным, нужно делать константным